builtin/providers/aws/provider.go

```diff
@@ -113,6 +113,7 @@ func Provider() terraform.ResourceProvider {
 		DataSourcesMap: map[string]*schema.Resource{
 			"aws_ami":                dataSourceAwsAmi(),
 			"aws_availability_zones": dataSourceAwsAvailabilityZones(),
+			"aws_s3_object":          dataSourceAwsS3Object(),
```

Comment:
Sure, that's no problem at all, I can rename it.

Generally speaking though I'd rather see resource called `aws_s3_object` as it just makes more sense when you pronounce it (IMO). 😃 

I assume that consistency with past choices is probably easier & safer to implement than deprecating past choices though.